pkg/driver/controller.go

```diff
@@ -508,11 +510,27 @@ func (d *controllerService) ControllerExpandVolume(ctx context.Context, req *csi
 		return nil, status.Error(codes.InvalidArgument, "After round-up, volume size exceeds the limit specified")
 	}
 
-	actualSizeGiB, err := d.cloud.ResizeDisk(ctx, volumeID, newSize)
+	modifyVolumeRequest := modifyVolumeRequest{
+		newSize: newSize,
+	}
+
+	err := d.addModifyVolumeRequest(ctx, volumeID, &modifyVolumeRequest)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not resize volume %q: %v", volumeID, err)
 	}
 
+	var actualSizeGiB int64
+
+	select {
+	case err := <-d.modifyVolumeManager.requestHandlerMap[volumeID].errChan:
+		d.modifyVolumeManager.deleteRequestHandler(volumeID)
+		return nil, status.Errorf(codes.Internal, "Could not resize volume %q: %v", volumeID, err)
+	case size := <-d.modifyVolumeManager.requestHandlerMap[volumeID].resultChan:
+		actualSizeGiB = size
+	}
```

Comment:
(see my comment in `startVolumeTimer` for more context)

This should have an extra case for the context expiring.